sessionctx/variable/session.go

```diff
@@ -179,6 +187,16 @@ func (tc *TransactionContext) SetForUpdateTS(forUpdateTS uint64) {
 	}
 }
 
+// SetStmtFuture sets the stmtFuture .
+func (tc *TransactionContext) SetStmtFuture(future oracle.Future, cachedTS uint64) {
+	tc.stmtFuture = &stmtFuture{future: future, cachedTS: cachedTS}
+}
+
+// GetStmtFuture gets the stmtFuture.
+func (tc *TransactionContext) GetStmtFuture() (oracle.Future, uint64) {
+	return tc.stmtFuture.future, tc.stmtFuture.cachedTS
```

Comment:
Do we need to check if `tc.stmtFuture` is nil?

Reply:
It should not happen. We set the `stmtFuture` for every statement which needs to update the `forUpdateTS`. If it happened it should be a bug.